[OSCD] Added explorer.exe lolbin

caliskanfurkan opened the pull request:
explorer.exe LOLbin #40 in https://github.com/Neo23x0/sigma/issues/1014

caliskanfurkan reviewed (commented):
Fixed title

Neo23x0 commented on rules/windows/process_creation/win_susp_explorer.yml:
```diff
@@ -0,0 +1,32 @@
+            - \explorer.exe
+        CommandLine|contains:
+            - /root
+    selection2:
```
Doesn't this trigger every time an 'explorer.exe' gets invoked from a 'cmd.exe'?

caliskanfurkan replied:
Added as a known FP. Thank you.

yugoslavskiy replied:
Hello @caliskanfurkan!

Thank you for your contribution! 

It seems that we did a mistake while compiling the table in issue #1014. 
[A rule that covers the selection1 of your rule](https://github.com/Neo23x0/sigma/blob/master/rules/windows/process_creation/win_susp_explorer_break_proctree.yml) was already developed by @Neo23x0 back in June 2019.
We missed that, I am sorry for it.
Could you please remove section1 from your rule?
If you are OK with that, please put "low" to the `level`, as this detection is not going to produce high fidelity alerts, but more like events for triage and further dynamic severity calculation. 

Thank you!

yugoslavskiy replied:
Hello @caliskanfurkan! Could you please fix this?

caliskanfurkan replied:
> Hello @caliskanfurkan! Could you please fix this?

Done

caliskanfurkan reviewed (commented):
Added legit explorer.exe spawn from cmd.exe as false-positive

aw350m33d commented:
Hello, Furkan! Thanks for the pull request!

I have some suggestions for your rules:
1) Could you correct the date in the rules `9eb271b9-24ae-4cd4-9465-19cfc1047f3e`, `bf6c39fc-e203-45b9-9538-05397c1b4f3f` and `bafac3d6-7de9-4dd9-8874-4a1194b493ed` according to the guide (https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#date)

2) In the rule `bf6c39fc-e203-45b9-9538-05397c1b4f3f` I suggest using the `all` modifier. 
Something like this:
```
selection_V_L:
    CommandLine | contains | all:
        - / V
        - / L
```
Then the condition will be simpler:
```
selectionFindstr and (selection_V_L or selection_S_I)
```

3) The rule `bafac3d6-7de9-4dd9-8874-4a1194b493ed` does not explicitly indicate that a remote creation of a file is detected. This rule will also triggers for the commandline _`print.exe /D:PRN C:\somefile.txt`_, which is not what I would like to find with this rule.


Thank you for your active participation in the sprint from the very first days!

caliskanfurkan commented:
> Hello, Furkan! Thanks for the pull request!
> 
> I have some suggestions for your rules:
> 
> 1. Could you correct the date in the rules `9eb271b9-24ae-4cd4-9465-19cfc1047f3e`, `bf6c39fc-e203-45b9-9538-05397c1b4f3f` and `bafac3d6-7de9-4dd9-8874-4a1194b493ed` according to the guide (https://github.com/Neo23x0/sigma/wiki/Rule-Creation-Guide#date)
> 2. In the rule `bf6c39fc-e203-45b9-9538-05397c1b4f3f` I suggest using the `all` modifier.
>    Something like this:
> 
> ```
> selection_V_L:
>     CommandLine|contains|all:
>         - /V
>         - /L
> ```
> 
> Then the condition will be simpler:
> 
> ```
> selectionFindstr and (selection_V_L or selection_S_I)
> ```
> 
> 1. The rule `bafac3d6-7de9-4dd9-8874-4a1194b493ed` does not explicitly indicate that a remote creation of a file is detected. This rule will also triggers for the commandline _`print.exe /D:PRN C:\somefile.txt`_, which is not what I would like to find with this rule.
> 
> Thank you for your active participation in the sprint from the very first days!

Thank you. Fixed.

aw350m33d commented on rules/windows/registry_event/sysmon_comhijack_sdclt.yml:
```diff
@@ -0,0 +1,26 @@
+title: COM Hijack via Sdclt
+id: 07743f65-7ec9-404a-a519-913db7118a8d
```
This rule already pushed in repo (https://github.com/Neo23x0/sigma/blob/master/rules/windows/registry_event/sysmon_comhijack_sdclt.yml). Could you delete this rule from the PR?

aw350m33d commented:
> Thank you. Fixed.

Great!  

I think that rule `9eb271b9-24ae-4cd4-9465-19cfc1047f3e` we will merge with existing one.

And I still have a few more questions.

1) Rule `bf6c39fc-e203-45b9-9538-05397c1b4f3f` looks pretty good. But it has test status. Test rules need to be monitored in order to then make a conclusion about how efficient it is. Therefore, I propose to check it again and if it works, set the status to experimental.

2) As for rule `bafac3d6-7de9-4dd9-8874-4a1194b493ed`, could you give examples of situations (start lines) that you identify with this rule?

caliskanfurkan commented:
> 


> > Thank you. Fixed.
> 
> Great!
> 
> I think that rule `9eb271b9-24ae-4cd4-9465-19cfc1047f3e` we will merge with existing one.
> 
> And I still have a few more questions.
> 
> 1. Rule `bf6c39fc-e203-45b9-9538-05397c1b4f3f` looks pretty good. But it has test status. Test rules need to be monitored in order to then make a conclusion about how efficient it is. Therefore, I propose to check it again and if it works, set the status to experimental.

Fixed

> 2. As for rule `bafac3d6-7de9-4dd9-8874-4a1194b493ed`, could you give examples of situations (start lines) that you identify with this rule?

I tried like "print /D:C\....\.exe C:\exe" With this version, It doesn't create collusion with print.exe. But yes, using it like print.exe /D:C:\......txt will be a bypass for this rule (better than nothing IMHO) I also thought about excluding .txt from commandline but it won't catch #1 in LOLbin guide (ADS)

aw350m33d commented:
Hi Furkan! Thank you for your patience! 

I absolutely agree with you that having a rule is always better than not having one. 👍🏻  It just seems to me that we can configure it so that it will be difficult to bypass it. 💪🏻 

Yesterday I tried to make a regular expression for this case, but the expression turned out to be too long (but it covers all cases).
What do you think of this approach?

In general, your rule should detect what you described, but the rule has complex logic. I would like something simpler. Sorry for the importunity, I believe that we can do it simply and beautifully.

caliskanfurkan commented:
> Hi Furkan! Thank you for your patience!
> 
> I absolutely agree with you that having a rule is always better than not having one. 👍🏻 It just seems to me that we can configure it so that it will be difficult to bypass it. 💪🏻
> 
> Yesterday I tried to make a regular expression for this case, but the expression turned out to be too long (but it covers all cases).
> What do you think of this approach?
> 
> In general, your rule should detect what you described, but the rule has complex logic. I would like something simpler. Sorry for the importunity, I believe that we can do it simply and beautifully.

Hi, IMHO regex is overkill for a rule that only generates FP for cmdline printing (which I've never did) but doesn't matter for me. You can change.

aw350m33d commented:
Okay, I understand your point of view. Indeed, the likelihood of falls is not that great. 😃 

Then I propose to consider all this rules ready for merging. @yugoslavskiy 
Thank you very much for the discussion! 👍🏻

yugoslavskiy commented on rules/windows/process_creation/win_susp_explorer.yml:
```diff
@@ -0,0 +1,32 @@
+title: Proxy Execution Via Explorer.exe
+id: 9eb271b9-24ae-4cd4-9465-19cfc1047f3e
+description: Attackers can use explorer.exe for evading defense mechanisms
+author: 'Furkan CALISKAN, @caliskanfurkan_, OSCD Community'
```
could you please change "OSCD Community" to "oscd.ommunity" or "@oscd_initiative"? 
the thing is that oscd is an initiative, not a community.

the only place where you can see the "community" word near to the "oscd" word is the domain name.
but the only reason I've chosen this domain is that it was cheap and kinda matching the idea — it is a community activity of course, but the global infosec community, not the "oscd community". 

sorry for inconvenience.

caliskanfurkan replied:
Done

yugoslavskiy commented on rules/windows/process_creation/win_susp_findstr.yml:
```diff
@@ -0,0 +1,32 @@
+title: Abusing Findstr for Defense Evasion 
+id: bf6c39fc-e203-45b9-9538-05397c1b4f3f
+description: Attackers can use findstr to hide their artifacts or search specific strings and evade defense mechanism
+author: 'Furkan CALISKAN, @caliskanfurkan_, OSCD Community'
```
could you please also put "oscd.community" here?

caliskanfurkan replied:
Done

yugoslavskiy commented:
Thank you very much, @caliskanfurkan! Well done!

yugoslavskiy reviewed (approved):
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.
